zebra-consensus/src/block/subsidy/general.rs

```diff
@@ -102,6 +102,26 @@ pub fn find_output_with_amount(
         .collect()
 }
 
+/// Validate shielded consensus rules as described in [ZIP-213][ZIP-213]
+///
+/// [ZIP-213]: https://zips.z.cash/zip-0213#specification
+pub fn shielded_coinbase(height: Height, network: Network, transaction: &Transaction) -> bool {
+    let heartwood_height = Heartwood
+        .activation_height(network)
+        .expect("heartwood activation height should be available");
+
+    if height < heartwood_height {
+        if !transaction.joinsplits() && transaction.shields().is_none() {
+            return true;
+        }
+    } else {
+        if !transaction.joinsplits() {
```

Comment:
Same as above. After https://github.com/ZcashFoundation/zebra/pull/1190/files#r508904039 is fixed this needs to be changed to `transaction.joinsplits().is_none()`